text/1185771/higham_przytycka.tex

```diff
@@ -0,0 +1,245 @@
+
+\usepackage{lmodern}
+
+\newtheorem{theorem}{Theorem}[section]
+
+\newtheorem{lemma}{Lemma}
+\newtheorem{corollary}{Corollary}
+
+\title{A summary and modification of Higham \& Przytycka's algorithm for maximum finding on rings, for the undirected case}
+\author{Jakub Migdał}
+\date{}
+
+\begin{document}
+\maketitle
+
+\section{A starting point}
+We begin by proposing a simple, round-based leader election scheme on directed rings which we will then extend with two heuristics that lead to an improvement in the number of rounds and sent messages.
+\\
+\begin{algorithmic}
+\Function{DropMessage}{}
+    \State \Return received.round = last.round and
+    \State \ \ \ \ ((odd(received.round) and received.id \textgreater \ last.id) or
+    \State \ \ \ \  (even(received.round) and received.id \textless   \ last.id))
+\EndFunction
+\\
+
+\Function{Promotion}{}
+    \State \Return received.round = last.round
+\EndFunction
+\\
+
+\Function{IsLeader}{}
+    \State \Return received.round = last.round and received.id = last.id
+\EndFunction
+\\
+\end{algorithmic}
+
+\begin{algorithm}[H]
+\caption{LE.BASIC}
+\begin{algorithmic}
+\Repeat
+    \If{\Call{DropMessage}{} is false} 
+        \If{\Call{Promotion}{} is true}
+            \State $received.round \gets received.round+1$
+        \EndIf
+        \State $last \gets received$
+        \State \Call{send}{received}
+    \EndIf
+\Until \Call{IsLeader}{}
+\end{algorithmic}
+\end{algorithm}
+
+From a high-level perspective the algorithm works as follows: each node sends an envelope containing its ID, and saves it as the last envelope it has sent. The nodes (referred to as processors in the paper, a nomenclature we too shall use) then act as actors comparing received envelopes to the last seen envelope, and eliminating them if the defined conditions hold. This effectively produces a "chain" reaction where each round, the surviving envelopes are gathered into maximal chains of descending (ascending, depending on the round's parity) ID's, from which only the minimum (maximum) of its chain remains.
+\\\\\\
+\newpage
+This perspective immediately establishes the algorithm's correctness based on the following properties:
+\\\\
+1. Safety: It will never delete all envelopes \\
+\\
+2. Progress: Only one envelope will remain, since a sequence of two alternating rounds must drop a non-zero number of envelopes \\
+\\
+3. Correct termination: The algorithm terminates when a processor detects the received envelope is the one it has last sent, meaning it has made a round trip uninterrupted. \\
+\\
+It can be shown the algorithm uses no more than $n\log_{\phi}{n} \leq 1.44n$ messages, based on the following property:\\
+
+\begin{lemma}
+An envelope in round $i$ travels a distance of at least $F_{i+1}$ before advancing to the next round (where $F_i$ is the i'th Fibonacci number)
+\end{lemma}
+The proof follows by induction and considering a few cases which we will omit for the sake of the modified algorithm.
```

Comment:
Please provide an example case, I completely don't see why it should follow.

Reply:
Upon closer inspection, I believe this to be a typo in the paper, since the following lemmas either prove a specific case of this exact property (Lemma 4), or a weaker bound (Lemma 5). Though it mentions an equivalent algorithm by Peterson achieving the exact same bound, but the analysis is a bit more involved.

Reply:
OK, that suffices, thank you